pytorch_toolkit/formula_recognition/tools/export.py

```
@@ -0,0 +1,273 @@
+from im2latex.data.vocab import START_TOKEN, END_TOKEN, read_vocab
+from im2latex.models.im2latex_model import Im2latexModel
+
+
+def read_net(model_xml, ie, device):
+    model_bin = os.path.splitext(model_xml)[0] + ".bin"
+    model = ie.read_network(model_xml, model_bin)
+
+    return model
+
+
+class ONNXExporter():
+    def __init__(self, config):
+        self.config = config
+        self.model_path = config.get('model_path')
+        self.vocab = read_vocab(config.get('vocab_path'))
+        self.transform = create_list_of_transforms(config.get('transforms_list'))
+        self.transform_for_ir = create_list_of_transforms(config.get('transforms_list'), ovino_ir=True)
+        self.model = Im2latexModel(config.get('backbone_type'), config.get(
+            'backbone_config'), len(self.vocab), config.get('head'))
+        if self.model_path is not None:
+            self.model.load_weights(self.model_path, old_model=config.get("old_model"))
+
+        self.input = config.get("dummy_input")
+        self.img = self.read_and_preprocess_img(self.transform)
+        self.img_for_ir = self.read_and_preprocess_img_for_ir(self.transform_for_ir)
+        self.encoder = self.model.get_encoder_wrapper(self.model)
+        self.decoder = self.model.get_decoder_wrapper(self.model)
+
+    def read_and_preprocess_img(self, transform):
+        img = cv.imread(self.input)
+        img = transform(img)
+        img = img[0].unsqueeze(0)
+        return img
+
+
+    def read_and_preprocess_img_for_ir(self, transform):
+        img = cv.imread(self.input)
+        img = transform(img)
+        img = np.expand_dims(img[0], axis=0)
+        return np.transpose(img, (0, 3, 1, 2))
+
+    def export_encoder(self):
+        torch.onnx.export(self.encoder, self.img, self.config.get("res_encoder_name"),
+                          opset_version=11,
+                          input_names=self.config.get("encoder_input_names").split(','),
```

Sorry, but why do you get `"encoder_input_names"`, `"encoder_output_names"`, etc from the config file?

As far as I can see, they depend on the `EncoderWrapper` and `DecoderWrapper` -- particularly, they depend on what they return by the `forward` method.

So, it cannot be tweaked by the config file.

Moreover, as far as I can see from the next lines, the order of the fields are predefined (shape of the first encoder output name should be `(batch, H, W)`.

So, if the config file turn out to be not compliant w.r.t. the code we will get error (or a silent error, which we will see in demo only). The only thing that may be changed in the config file is the names of the outputs. But (IMHO) it is better to use predefined names here (to be compliant with OMZ demo).

So, I'd propose to remove the fields from the config file.

---

Made name optional in the config file. They are still got from a config file but there are also named constants in the export script in case `.get()` method returns None from a config.